Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -223,24 +217,20 @@ protected override void Dispose(bool isDisposing)
             TryExecute(() =>
             {
                 _disposeCalled = true;
-                lock (_inUseConnections)
+                foreach (var inUseConnection in _inUseConnections)
                 {
-                    var connections = new List<IPooledConnection>(_inUseConnections.Values);
-                    _inUseConnections.Clear();
-                    foreach (var inUseConnection in connections)
+                    Logger?.Info($"Disposing In Use Connection {inUseConnection.Id}");
+                    if (_inUseConnections.TryRemove(inUseConnection))
                     {
-                        Logger?.Info($"Disposing In Use Connection {inUseConnection.Id}");
                         inUseConnection.Close();
```

Comment:
In theory, no, as all impl of `IConnection.Close` method will only log errors but not throw any out.